### ISSUE TITLE
No warning is issued against `[ ! -a p ]` and `test ! -a p`

### ISSUE THREAD

**Issue by e-kwsm** [koalaman/shellcheck#3174]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2166?
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
# or
#!/bin/ksh
[ ! -a /bin/sh ]
echo $?
```

#### Here's what shellcheck currently says:

nothing

#### Here's what I wanted or expected to see:

A warning is issued against `[ ! -a p ]` since it is not well-defined.
Actually, bash 5.2.37 and ksh 2020.0.0 give different results: 0 and 1, respectively. It seems that bash performs
```bash
[ ! ] && [ /bin/sh ]
# or
[ -n '!' ] && [ -n /bin/sh ]
```
whereas ksh does
```ksh
! [ -a /bin/sh ]
# or
! [ -e /bin/sh ]
```

**Comment by koalaman** [comment]
ShellCheck now generally recommends `-e` over unary `-a`, warns about `[ -o opt ]` not being POSIX when applicable, and about the pitfall of `[ ! -o braceexpand ]`. Very surprising behavior and a great check, thanks!
